homeassistant/components/crownstone/light.py

```diff
@@ -0,0 +1,206 @@
+            return SUPPORT_BRIGHTNESS
+        return 0
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """State attributes for Crownstone devices."""
+        attributes: dict[str, Any] = {}
+        # switch method
+        if self.usb is not None and self.usb.is_ready():
+            attributes["Switch Method"] = "Crownstone USB Dongle"
+        else:
+            attributes["Switch Method"] = "Crownstone Cloud"
+
+        # crownstone abilities
+        attributes["Dimming"] = ABILITY_STATE.get(
+            self.device.abilities.get(DIMMING_ABILITY).is_enabled
+        )
+        attributes["Tap To Toggle"] = ABILITY_STATE.get(
+            self.device.abilities.get(TAP_TO_TOGGLE_ABILITY).is_enabled
+        )
+        attributes["Switchcraft"] = ABILITY_STATE.get(
+            self.device.abilities.get(SWITCHCRAFT_ABILITY).is_enabled
+        )
+
+        return attributes
+
+    async def async_added_to_hass(self) -> None:
+        """Set up a listener when this entity is added to HA."""
+        # new state received
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, SIG_CROWNSTONE_STATE_UPDATE, self.async_write_ha_state
+            )
+        )
+        # updates state attributes when usb connects/disconnects
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, SIG_UART_STATE_CHANGE, self.async_write_ha_state
+            )
+        )
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on this light via dongle or cloud."""
+        if ATTR_BRIGHTNESS in kwargs:
+            try:
+                # UART instance is not None when a USB is set up and in the same sphere
+                # UART instance is ready when USB is set up and plugged in
+                if self.usb is not None and self.usb.is_ready():
+                    await self.hass.async_add_executor_job(
+                        partial(
+                            self.usb.dim_crownstone,
+                            self.device.unique_id,
+                            hass_to_crownstone_state(kwargs[ATTR_BRIGHTNESS]),
+                        )
+                    )
+                else:
+                    await self.device.async_set_brightness(
+                        hass_to_crownstone_state(kwargs[ATTR_BRIGHTNESS])
+                    )
+                # set brightness
+                self.device.state = hass_to_crownstone_state(kwargs[ATTR_BRIGHTNESS])
+                self.async_write_ha_state()
+            except CrownstoneAbilityError as ability_error:
+                _LOGGER.error(ability_error)
+
+        elif self.usb is not None and self.usb.is_ready():
+            await self.hass.async_add_executor_job(
+                partial(self.usb.switch_crownstone, self.device.unique_id, on=True)
+            )
+            # set state (in case the updates never comes in)
```

Comment:
The above block is wrapped in a `try` but this block isn't.  Is that intentional?

Reply:
Yes, this one does not throw any exception, the uart packet is emitted into the mesh, and that's pretty much it. In future version we will likely wait for a reply and then we could raise `asyncio.TimeoutError` if we didn't get a reply in time for example. But that could always be added later for extra clarity on the state